### PR TITLE
test: protoassert in stores

### PR DESCRIPTION
**Pull request by janisz**
## Description

- #11531
- #11533
- #11503
- #11507
- #11575
- #11571
- #11572  
- https://github.com/stackrox/stackrox/pull/11582
- https://github.com/stackrox/stackrox/pull/11585
- https://github.com/stackrox/stackrox/pull/11588 
- https://github.com/stackrox/stackrox/pull/11583 :point_left:
- https://github.com/stackrox/stackrox/pull/11618 
- https://github.com/stackrox/stackrox/pull/11619 

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by mtodor — APPROVED** [review]
Looks good.

I didn't double-check all generated files. I looked at template changes and a few related files. If there is a need to look at some files that have `// Code generated by pg-bindings generator. DO NOT EDIT.` flag but are maintained manually, please list them and request re-review.

File: tools/generate-helpers/pg-table-bindings/singleton_test.go.tpl [hunk 1]
```diff
@@ -61,12 +62,12 @@ func (s *{{$namePrefix}}StoreSuite) TestStore() {
...
 	s.NoError(err)
 	s.True(exists)
-	s.Equal({{$name}}, found{{.TrimmedType|upperCamelCase}})
+	protoassert.Equal(s.T(), {{$name}}, found{{.TrimmedType|upperCamelCase}})
 
 	found{{.TrimmedType|upperCamelCase}}, exists, err = store.Get(ctx)
 	s.NoError(err)
 	s.True(exists)
-	s.Equal({{$name}}, found{{.TrimmedType|upperCamelCase}})
+	protoassert.Equal(s.T(), {{$name}}, found{{.TrimmedType|upperCamelCase}})
```
[the remark this answers is not in the document]
Comment (janisz) [reply tools/generate-helpers/pg-table-bindings/singleton_test.go.tpl:70]: Yes, I've seen that (tbh I was surprised that we have this test :) I think we need to clean this test e..g with subtest to add information what is tested as currently it's not easy to understand especially with copied assertions

[the remark this answers is not in the document]
Comment (janisz) [reply tools/generate-helpers/pg-table-bindings/singleton_test.go.tpl:70]: I'll create a separated PR for this

File: tools/generate-helpers/pg-table-bindings/singleton_test.go.tpl [hunk 2]
```diff
@@ -83,7 +84,7 @@ func (s *{{$namePrefix}}StoreSuite) TestStore() {
...
```
[the remark this answers is not in the document]
Comment (janisz) [reply tools/generate-helpers/pg-table-bindings/singleton_test.go.tpl:87]: It's fixed

File: tools/generate-helpers/pg-table-bindings/singleton_test.go.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (janisz) [reply tools/generate-helpers/pg-table-bindings/singleton_test.go.tpl:70]: - https://github.com/stackrox/stackrox/pull/11690

**Review by rhybrillou — COMMENTED** [review]
LGTM modulo rebase to remove changes from #11582

**Pull request merged**